File: app/src/main/java/cosmic/com/pkwprj/model/Office.java
package cosmic.com.progressbartest;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;

public class Office implements Serializable {

    @SerializedName( "name" ) private String name;
    @SerializedName( "location" ) private String location;
    @SerializedName( "reservations" )private ArrayList<Reservations> reservations;

    public Office(String name, String location) {
        this.name = name;
        this.location = location;
    }

    public Office(String name, String location, ArrayList<Reservations> reservations) {
        this.name = name;
        this.location = location;
        this.reservations = reservations;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }


    public class Reservations implements Serializable{
        @SerializedName( "startTime" )
        private String startTime;
        @SerializedName( "endTime" )
        private String endTime;

        public Reservations(String startTime, String endTime) {
            this.startTime = startTime;
            this.endTime = endTime;
        }

        public String getStartTime() {
            return startTime;
        }

        public void setStartTime(String startTime) {
            this.startTime = startTime;
        }

        public String getEndTime() {
            return endTime;
        }

        public void setEndTime(String endTime) {
            this.endTime = endTime;
        }
    }


    public ArrayList<Reservations> getReservations() {
        return reservations;
    }

    public void setReservations(ArrayList<Reservations> reservations) {
        this.reservations = reservations;
    }
}

File: app/src/main/java/cosmic/com/pkwprj/model/Office.kt
package cosmic.com.pkwprj.model

import android.graphics.drawable.Drawable
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.*

class Office : Serializable {

    @SerializedName("name")
    var name: String? = null
    @SerializedName("location")
    var location: String? = null
    @SerializedName("reservations")
    var reservations: ArrayList<Reservations>? = null
    @SerializedName("drawable")
    var drawable: Drawable? = null

    constructor(name: String, location: String) {
        this.name = name
        this.location = location
    }

    constructor(
        name: String?,
        location: String?,
        reservations: ArrayList<Reservations>?,
        drawable: Drawable
    ) {
        this.name = name
        this.location = location
        this.reservations = reservations
        this.drawable = drawable
    }


    inner class Reservations(
        @field:SerializedName("startTime")
        var startTime: String?, @field:SerializedName("endTime")
        var endTime: String?
    ) : Serializable
}
